Documentation/Functions/Typolink.rst

```diff
@@ -543,27 +543,66 @@ userFunc
     :name: typolink-userFunc
     :type: :ref:`data-type-function-name`
 
-    This passes the link-data compiled by the typolink function to a user-
-    defined function for final manipulation.
+    As `userFunc` is the last property of `typolink`, all previous properties
+    have already been processed into an object of type `LinkResultInterface`,
+    which is passed as the first parameter to the defined `userFunc` for
+    further manipulation.
 
-    The :php:`$content` variable passed to the user-function (first parameter) is
-    an array with the keys "TYPE", "TAG", "url", "targetParams" and
-    "aTagParams".
+    All `typolink` properties are passed as an array as the second parameter.
 
-    TYPE is an indication of link-kind: mailto, url, file, page
+    The current request object incl. current instance of `ContentObjectRenderer`
+    will be passed as third parameter.
 
-    TAG is the full <A>-tag as generated and ready from the typolink
-    function.
+    As the `userFunc` property must also return an object of type
+    `LinkResultInterface`, it makes sense to use this LinkResult object
+    passed and customise it to suit your needs.
 
-    The actual tag value is constructed like this:
+    ..  code-block:: typoscript
+
+        # Register your UserFunc with TypoScript
+        lib.parseFunc_RTE.userFunc = MyVendor\MySitePackage\UserFunc\ModifyTypoLinkUserFunc->modifyTypoLinkResult
+
+    The following PHP example checks whether the object passed contains a link
+    of type `page` (this is the value behind: `LinkService::TYPE_PAGE`). If yes,
+    the link is opened in a new browser tab, the link title is set to the page
+    title and the link text will be wrapped in :html:`<strong>` tags.
 
     ..  code-block:: php
```

Comment:
I would suggest to use a literalinclude here: https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/WritingReST/Reference/Code/Codeblocks.html#literalinclude

This way we have the php in its own file, it can be linted and cgls automatically applied. Also please add a caption about where the file is located in a project.